### PR TITLE
lb maps rotation

### LINKED ISSUE #14: Rotate the LiteBIRD CMB maps in get_lb_map
The noise and foreground for CMB-S4 are in Equatorial coordinates while for LiteBIRD are in Galactic coordinates. The CMB maps fetched through `maps` package needs a relative rotation between the LiteBIRD and CMB-S4 CMB maps. This is cheaper to implement on the LiteBIRD maps as they are at a lower resolution (hence lower `lmax`). We need a `rotate_alm` stage in `get_lb_map` to convert from Equatorial to Galactic coordinates.

@antolonappan Please add any requirements that I may have missed.

### PR DESCRIPTION
**Pull request by antolonappan**
Added the option for lmax and coordinate rotation to `maps._build_maps`.  This also set the S4 and LB maps to equatorial and galactic coordinate system respectively. Example script is shown in the notebook

### CODE REVIEW

**Comment by antolonappan** [comment]
Hi @carronj , 

Your username is not listing in the reviewers panel, so I couldn't send you a review request. But please have a look at this PR.

Thanks,
Anto.

File: 20231018_signal_cmb/maps.py [hunk 1]
```diff
@@ -30,10 +30,13 @@
 except:
     print('maps.py: could not read LiteBird instrument model at ' + path2lb)
     lb = None
-def _build_maps(idx, beam_amin: float, nside:int, job='TQU'):
+def _build_maps(idx, beam_amin: float, nside:int, job='TQU',lmax=4096,coord=None):
```
Comment (zonca) [inline 20231018_signal_cmb/maps.py]: can you run `black` to fix formatting?

File: 20231018_signal_cmb/maps.py [hunk 2]
```diff
@@ -48,11 +51,15 @@ def _build_maps(idx, beam_amin: float, nside:int, job='TQU'):
...
-        maps[0] = hp.alm2map(teb[0], nside)
+        maps_ = hp.alm2map(teb[0], nside)
+        maps[0] = rot.rotate_map_pixel(maps_) if coord else maps_
+        del maps_
     if 'QU' in job.upper():
         hp.almxfl(teb[1], bl_E * pw_P, inplace=True)
         hp.almxfl(teb[2], bl_B * pw_P, inplace=True)
-        maps['T' in job.upper():] = hp.alm2map_spin(teb[1:], nside, 2, lmax)
+        maps_ = hp.alm2map(teb, nside)
+        maps['T' in job.upper():] = rot.rotate_map_pixel(maps_)[1:] if coord else maps_[1:]
```
Comment (zonca) [inline 20231018_signal_cmb/maps.py]: any reason you are using pixel space rotation?
we are already in Alm space, we could do a `rotate_alm`

File: 20231018_signal_cmb/maps.py (hunk 1 above)
Comment (zonca) [inline 20231018_signal_cmb/maps.py]: please add types for the new arguments

Comment (zonca) [inline 20231018_signal_cmb/maps.py]: I see originally there was no docstring, could you add it?

**Review by zonca — CHANGES_REQUESTED** [review]
thanks for taking care of this! some improvements

File: 20231018_signal_cmb/maps.py (hunk 2 above)
Comment (antolonappan) [reply 20231018_signal_cmb/maps.py]: I will try to change it to alms, I just realised that, rotate_alm has the option for lmax

File: 20231018_signal_cmb/maps.py (hunk 1 above)
Comment (antolonappan) [reply 20231018_signal_cmb/maps.py]: yes

Comment (antolonappan) [reply 20231018_signal_cmb/maps.py]: ok

**Comment by antolonappan** [comment]
- Current rotation is based on alms, because pixel based rotation was giving a slightly different power in the high-l s. 
- Run `black` 
- Added docstring 

Current routine is selected based on the test done [here](https://gist.github.com/antolonappan/0d5a6159d022f9ef57f041f5f3d95d46)

The current version converts alm to  map and back to alm, because the `lmax` argument in `rotate_alm` was terminating the kernel. 
lmax_crashing

**Comment by carronj** [comment]
Why not just copying the alm array to lmax instead of back and forth with alm2map and map2alm ? (that's not something you really want to do on healpix grid.   )

**Comment by antolonappan** [comment]
@carronj Yes, I will do that

File: 20231018_signal_cmb/maps.py [hunk 3]
```diff
@@ -14,78 +15,133 @@
...
     if 'T' in job.upper():
         hp.almxfl(teb[0], bl_T * pw_T, inplace=True)
+        if coord:
+            rot.rotate_alm(teb[0], inplace=True)
         maps[0] = hp.alm2map(teb[0], nside)
     if 'QU' in job.upper():
         hp.almxfl(teb[1], bl_E * pw_P, inplace=True)
         hp.almxfl(teb[2], bl_B * pw_P, inplace=True)
+        if coord:
+            rot.rotate_alm(teb[1], inplace=True)
```
Comment (zonca) [inline 20231018_signal_cmb/maps.py]: I think you need to rotate TEB all together, so polarization is handled properly by HEALPix

**Review by zonca — COMMENTED** [review]
Do people often do QU? I am worried rotate_alm would either not work or consider the alms like 2 temperature alms. Worth checking, otherwise you need to create a zero temperature map for that case.

**Comment by antolonappan** [comment]
The current version rotate the alms all together irrespective of the `job` argument value. So even QU is used, it should work.

**Review by zonca — APPROVED** [review]

**Pull request merged**